server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1383,7 +1393,13 @@ public MergeStats mergeStats() {
         if (engine == null) {
             return new MergeStats();
         }
-        return engine.getMergeStats();
+
+        final MergeStats mergeStats = engine.getMergeStats();
+        if (indexSettings.shouldCleanupUnreferencedFiles()) {
```

Comment:
Is `shouldCleanupUnreferencedFiles` a dynamic setting? If yes and if its value is false, response would not show the stats at all. Is this understanding correct? 

IMO, we should always show the response with 0 value as the caller of the API can break if some values are present and missing based on a dynamic setting.

Reply:
I think it is to protect compatiblity issue. A similar thing is done for Remote store: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L1393